### DISCUSSION TITLE
Using firmware on Ender 3 Neo, Ender 3 V2 Neo, Ender 3 S1

### DISCUSSION THREAD

**Discussion by zforged** [mriscoc/Ender3V2S1#1330] [Q&A]
Hello,

I'm interested in trying out this firmware on the machines metioned in title. However, is it possible to make firmware/printer modifications so that they work with manual mesh leveling, no stock CR touch or other probes? what are the changes I need to make so yall functions properly?

**Comment by mriscoc** [comment]
Hi,
If you want to use the Manual Mesh versions, the printer must have a Z-end stop switch properly installed.

**Reply by zforged** [reply]
Hi mriscoc!

May you elaborate what kind of endstops are available for Ender 3 series and how to properly install it?

I'm assuming using one like stock X and Y axis endstops should work but I have no idea how to install them. Specifically, S1 printer comes with dual Z axis, unlike other printers which only uses one Z axis. Does that mean for the S1, I need 2x Z endstops, or will 1 just do the job?

**Reply by zforged** [reply]
Hello mriscoc,

I just installed your firmware on one of my printers and will be configuring and playing with settings shortly. So far I think it's a great firmware, I like customization, its ease of use and installing. 

Could you assist me on how to use and install other types or sensors (eg. bl touch, capacitive) on ender 3 s1, and ender 3 neo printers, and will any of your firmware versions work with those sensors?

**Reply by mriscoc** [reply]
You can try first with Marlin, if Marlin supports your sensor then the Professional Firmware will also support it.

**Reply by zforged** [reply]
> You can try first with Marlin, if Marlin supports your sensor then the Professional Firmware will also support it.

I got the firmware running on Ender 3 V2 Neo, it's great,  but I can't seem to make it work on Ender 3 S1. My CR touch stopped working on this printer and I still cant figure out what's the cause/solution.  It doesn't work neither with Creality stock firmware nor with Professional Firmware. When prompted to initiate ABL, using Professional Firmware, the printer homes X and Y axis, then gets stuck at the center of the print bed and CR touch changes light display to red. I don't know when/why this started happening. May you have any suggestion about this?

**Reply by mriscoc** [reply]
Check the wiring, especially the white wire.
